README.md
Filesystem Resize Cookbook
==========================
[![Cookbook Version](https://img.shields.io/cookbook/v/filesystem_resize.svg?style=flat)](https://supermarket.chef.io/cookbooks/filesystem_resize)
[![Dependency Status](http://img.shields.io/gemnasium/zuazo/filesystem_resize-cookbook.svg?style=flat)](https://gemnasium.com/zuazo/filesystem_resize-cookbook)
[![Code Climate](http://img.shields.io/codeclimate/github/zuazo/filesystem_resize-cookbook.svg?style=flat)](https://codeclimate.com/github/zuazo/filesystem_resize-cookbook)
[![Build Status](http://img.shields.io/travis/zuazo/filesystem_resize-cookbook.svg?style=flat)](https://travis-ci.org/zuazo/filesystem_resize-cookbook)

This [Chef](https://www.chef.io/) cookbook resizes the file system automatically when the underlying partition or disk increases its size.

It is mainly oriented to work with cloud or virtual servers where it is common to change the disk size.

Requirements
============

## Platform Requirements

This cookbook has been tested on the following platforms:

* Amazon (>= 2012.03)
* CentOS (>= 6.0)
* Debian (>= 7.0)
* Fedora
* RedHat
* Ubuntu (>= 12.04)

Please, [let us know](https://github.com/zuazo/filesystem_resize-cookbook/issues/new?title=I%20have%20used%20it%20successfully%20on%20...) if you use it successfully on any other platform.

## Application Requirements

* Ruby 1.9.3 or higher.

The other required applications usually come with the operating system:

* `lsblk`, `findmnt` and `losetup`: included inside **[util-linux](http://en.wikipedia.org/wiki/Util-linux) (&ge; 2.19)** package.
* `pgrep`: included inside [procps-ng](http://sourceforge.net/projects/procps-ng/) package.
* `e2fsck`, `dumpe2fs` and `resize2fs` for *ext3* and *ext4*: included inside [e2fsprogs](http://e2fsprogs.sourceforge.net/) package.
* `xfs_info` and `xfs_growfs` for *XFS*: included inside [xfsprogs](http://oss.sgi.com/projects/xfs/) package.

Attributes
==========

| Parameter                                  | Default | Description                              |
|:-------------------------------------------|:--------|:-----------------------------------------|
| `node['filesystem_resize']['compiletime']` | `false` | Resize the file systems at compile time.

Recipes
=======

## filesystem_resize::default

Resizes all mounted file systems.

Resources
=========

## filesystem_resize(device)

Resizes a partition.

### filesystem_resize Actions

* `run` (default)

### filesystem_resize Parameters

| Parameter | Default           | Description                              |
|:----------|:------------------|:-----------------------------------------|
| device    | *resource name*   | Device full path.

## filesystem_resize_all(name)

Resizes all mounted file systems.

### filesystem_resize Actions

* `run` (default)

Usage
=====

## Including in a Cookbook Recipe

You can simply include it in a recipe:

```ruby
# in your recipe
include_recipe 'filesystem_resize'
```

Don't forget to include the `filesystem_resize` cookbook as a dependency in the metadata:

```ruby
# metadata.rb
depends 'filesystem_resize'
```

## Including in the Run List

Another alternative is to include it in your Run List:

```json
{
  "name": "app001.example.com",
  [...]
  "run_list": [
    [...]
    "recipe[filesystem_resize]"
  ]
}
```

Testing
=======

See [TESTING.md](https://github.com/zuazo/filesystem_resize-cookbook/blob/master/TESTING.md).

## ChefSpec Matchers

### filesystem_resize(device)

Helper method for locating a `filesystem_resize` resource in the collection.

```ruby
resource = chef_run.filesystem_resize('/dev/sda1')
expect(resource).to notify('service[apache2]').to(:restart)
```

### run_filesystem_resize(device)

Assert that the Chef Run runs `filesystem_resize`.

```ruby
expect(chef_run).to run_filesystem_resize('/dev/sda1')
```

### filesystem_resize_all(name)

Helper method for locating a `filesystem_resize_all` resource in the collection.

```ruby
resource = chef_run.filesystem_resize_all('default')
expect(resource).to notify('service[apache2]').to(:restart)
```

### run_filesystem_resize_all(name)

Assert that the Chef Run runs `filesystem_resize`.

```ruby
expect(chef_run).to run_filesystem_resize_all('default')
```

Contributing
============

Please do not hesitate to [open an issue](https://github.com/zuazo/filesystem_resize-cookbook/issues/new) with any questions or problems.

See [CONTRIBUTING.md](https://github.com/zuazo/filesystem_resize-cookbook/blob/master/CONTRIBUTING.md).

TODO
====

See [TODO.md](https://github.com/zuazo/filesystem_resize-cookbook/blob/master/TODO.md).

License and Author
==================

|                      |                                          |
|:---------------------|:-----------------------------------------|
| **Author:**          | [Xabier de Zuazo](https://github.com/zuazo) (<xabier@zuazo.org>)
| **Copyright:**       | Copyright (c) 2015, Xabier de Zuazo
| **Copyright:**       | Copyright (c) 2014-2015, Onddo Labs, SL.
| **License:**         | Apache License, Version 2.0

    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at
    
        http://www.apache.org/licenses/LICENSE-2.0
    
    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.


CHANGELOG.md
CHANGELOG for filesystem_resize
===============================

This file is used to list changes made in each version of the `filesystem_resize` cookbook.

## v0.3.0 (2015-09-04)

* Update chef links to use *chef.io* domain.
* Update contact information and links after migration.
* Update RuboCop to version `0.33.0`.
* metadata: Add `source_url` and `issues_url` links.

* Documentation:
 * README: Improve description.

* Testing:
 * Update Gemfile dependencies and Kitchen platforms.
 * Move ChefSpec tests to *test/unit*.
 * Use `ChefSpec::SoloRunner` to make unit tests faster.
 * Integrate tests with `should_not` gem.

## v0.2.0 (2015-01-23)

* Fix always resize bug.
* Add `filesystem_resize` and `filesystem_resize_all` resources, big refactor.
* Update Gemfile deps, update RuboCop.

## v0.1.0 (2014-08-11)

* Initial release of `filesystem_resize`.


metadata.rb
# encoding: UTF-8
#
# Cookbook Name:: filesystem_resize
# Author:: Xabier de Zuazo (<xabier@zuazo.org>)
# Copyright:: Copyright (c) 2014-2015 Onddo Labs, SL.
# License:: Apache License, Version 2.0
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

name 'filesystem_resize'
maintainer 'Xabier de Zuazo'
maintainer_email 'xabier@zuazo.org'
license 'Apache 2.0'
description <<-EOS
Resize the file system automatically when the underlying partition or disk
increases its size.
EOS
long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
version '0.4.0' # WiP

if respond_to?(:source_url)
  source_url "https://github.com/zuazo/#{name}-cookbook"
end
if respond_to?(:issues_url)
  issues_url "https://github.com/zuazo/#{name}-cookbook/issues"
end

supports 'amazon', '>= 2012.03'
supports 'centos', '>= 6.0'
supports 'debian', '>= 7.0'
supports 'fedora'
supports 'redhat'
supports 'ubuntu', '>= 12.04'

recipe 'filesystem_resize::default', 'Resizes all mounted file systems.'

provides 'filesystem_resize'
provides 'filesystem_resize_all'

attribute 'filesystem_resize/compiletime',
          display_name: 'fs resize at compile time',
          description: 'Resize the file sistems at compile time.',
          choice: %w(true false),
          type: 'string',
          required: 'optional',
          default: 'false'
